crates/ruff_linter/resources/test/fixtures/pylint/self_cls_assignment.py

```diff
@@ -0,0 +1,18 @@
+class Fruit:
+    @classmethod
+    def list_fruits(cls) -> None:
```

Comment:
Can we add some test cases which uses other type of functions? For example:

Using a normal function:
```py
def list_fruits(self, cls) -> None:
    self = "apple"  # Ok
    cls = "banana"  # Ok
```

Using a static method:
```py
class Fruit:
	@staticmethod
	def list_fruits(self, cls) -> None:
	    self = "apple"  # Ok
	    cls = "banana"  # Ok
```